### ISSUE TITLE
Misleading warning "is not valid under any of the given schemas" when linting tasks

### ISSUE THREAD

**Issue by boutetnico** [ansible/ansible-lint#2888]
##### Summary

I have a task file that is imported by a top-level playbook.
Ansible-lint reports a warning when checking the following file whereas I believe it is mistaken because the task works as expected. I suspect the `environment` key to cause the warning.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.10.2 using ansible 2.14.1
```

- ansible installation method: pip
- ansible-lint installation method: pip
- OS: macOS 12.6.1

##### STEPS TO REPRODUCE

Create the following `test.yml` file inside a `tasks` folder.
```yaml
---
- name: Run npm run prod
  ansible.builtin.command: npm run prod
  environment:
    DISABLE_V8_COMPILE_CACHE: 1
  changed_when: false
```

Run ansible-lint:
```bash
ansible-lint tasks/test.yml
```

##### Desired Behavior

No warning is issued.

##### Actual Behavior

A warning is issued.
![ansible-lint-warning](https://user-images.githubusercontent.com/973739/212069008-22a6a782-010c-49f8-be9c-04232580cd89.png)

**Comment by alqu** [comment]
I am running ansible-lint v6.18.0 and have exactly the same bug. So not solved:


**STEPS TO REPRODUCE**
```yaml
- name:         Check some command
  ansible.builtin.command:
    cmd:        "someCommand"
  environment:
    my_env_var: 1
  register:     command_result
  failed_when:  command_result.rc > 2
  changed_when: command_result.stdout != ''
```


**Desired Behavior**

No warning is issued.


**Actual Behavior**

A warning is issued.
```
WARNING  Listing 1 violation(s) that are fatal
schema[tasks]: {'name': 'Check some command', 'ansible.builtin.command': {'cmd': 'someCommand'}, 'environment': {'my_env_var': 1}, 'register': 'command_result', 'failed_when': 'command_result.rc > 2', 'changed_when': "command_result.stdout != ''"} is not valid under any of the given schemas
```
